src/VaVare/Statements/DeclarationStatement.cs

```diff
@@ -97,23 +167,29 @@ public LocalDeclarationStatementSyntax DeclareAndAssign(string name, Type type,
         /// <param name="arguments">Arguments to use when creating the variable.</param>
         /// <param name="useVarKeyword">Will use "var" keyword if true, otherwise the type.</param>
         /// <returns>The generated local declaration statement.</returns>
-        public LocalDeclarationStatementSyntax DeclareAndAssign(string name, Type type, ArgumentListSyntax arguments, bool useVarKeyword = true)
+        [Obsolete("Use the overload without optional parameter or without type instead")]
+        public LocalDeclarationStatementSyntax DeclareAndAssign(string name, Type type, ArgumentListSyntax arguments, bool useVarKeyword = true) 
+            => DeclareAndAssign(name, TypeGenerator.Create(type), arguments, useVarKeyword);
+
+        public LocalDeclarationStatementSyntax DeclareAndAssign(string name, TypeSyntax type, ArgumentListSyntax arguments, bool useVarKeyword = true)
         {
             if (string.IsNullOrEmpty(name))
```

Comment:
Ich weiß, dass kommt nicht von dir, aber hier sollte doch viel eher string.IsNullOrWhitespace hin, weil " " ist auch kein gültiger Name

Reply:
Ich habe gesehen, dass das an mehrere Stellen der Fall ist, eventuell muss man also mal übers ganze Projekt gehen und neue Unit Tests hinzufügen, die den Whitespace Fall mit abdecken

Reply:
Machen wir am besten in ner zweiten iteration. War jetzt ja eh erst mal nur das nötigste für den stream